### PR TITLE
added the files required for R bindings.

### PR DESCRIPTION
**Pull request by jbrowne6**
This incorporates the R bindings with Ubuntu makevars.  Without modification of makevars I don't think this will work on macs.

### CODE REVIEW

**Comment by falkben** [comment]
Can we have a few testthat tests to ensure bindings are working?

**Comment by MrAE** [comment]
I'm going to add install instructions to the README and tests to make sure the bindings are working.

File: R-Project/R/fpRerF.R [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+#'
+#'
+#' @export
+#'
+#' @examples
+#' library(rerf)
+#' forest <- fpRerF(as.matrix(iris[, 1:4]), as.numeric(iris[[5L]])-1)
+#'
+fpRerF <-
+	function(X=NULL, Y=NULL,csvFileName=NULL, columnWithY=NULL,minParent=1, numTreesInForest=100, numCores=1,numTreeBins=NULL, forestType="rerf", nodeSizeToBin=NULL, nodeSizeBin=NULL,mtry=NULL, mtryMult=NULL,seed=sample(1:1000000,1)){
```
Comment (falkben) [inline R-Project/R/fpRerF.R]: forestType should be `binnedBaseRerF` for default

File: R-Project/R/fpRerF.R [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		if(numCores < 1){
+			stop("at least one core must be used.")
+		}
+		if(minParent < 1){
+			stop("at least one observation must be used in each node.")
+		}
+		if(numTreesInForest < 1){
+			stop("at least one tree must be used.")
+		}
+		if(is.null(forestType)){
```
Comment (falkben) [inline R-Project/R/fpRerF.R]: This would only occur if someone passed NULL in for the forestType. Seems not needed.

File: R-Project/R/fpRerF.R [hunk 3]
```diff
@@ -0,0 +1,110 @@
+#' Packs a forest and saves modified forest to disk for use by PackPredict function
+#'
+#' Efficiently packs a forest trained with the RF option.  Two intermediate data structures are written to disk, forestPackTempFile.csv and traversalPackTempFile.csv.  The size of these data structures is proportional to a trained forest and training data respectively.  Both data structures are removed at the end of the operation.  The resulting forest is saved as forest.out.  The size of this file is similar to the size of the trained forest.
+#'
+#' @param X an n by d numeric matrix (preferable) or data frame used to train the forest. 
+#' @param Y a numeric vector of size n.  If the Y vector used to train the forest was not of type numeric then a simple call to as.numeric(Y) will suffice as input.
+#' @param csvFileName the name of a headerless csv file containing combined data and labels.
+#' @param columnWithY is the column in the headerless csv file containing class lables.
```
Comment (falkben) [inline R-Project/R/fpRerF.R]: may want to mention it's 0 indexed (what it is currenctly) or 1 indexed and adjust in the code with @MrAE's "fudge factor" :)

Comment (jbrowne6) [reply R-Project/R/fpRerF.R]: I'm all for starting the 0 index revolution in R.  Its about time they joined the rest of the programming community.

Comment (MrAE) [reply R-Project/R/fpRerF.R]: image

**Comment by falkben** [comment]
Also another todo:

now that maxDepth is a parameter we'll want to add it to the R binding.  Once this gets merged we'll then make sure to add parameters to both Python and R bindings whenever we add a feature to the C++ code

**Comment by MrAE** [comment]
I updated the README and the travis yaml file so that the RerF installs locally so that it can see the ../../packedForest directory.   

I commented out the `--check-as-cran` because this will cause travis to fail.

File: R-Project/R/fpRerF.R (hunk 3 above)
Comment (MrAE) [reply R-Project/R/fpRerF.R]: resolving for now, we'll add an issue for both python and R bindings to correct there.

**Comment by MrAE** [comment]
• I made a symbolic link to the packedForest directory from inside the `R-Project/src/` dir, so `R CMD build` now include it.

• I have updated the Makevars file and my user ~/.R/Makevars file so that `fpRerF` now works on Mac OS.  

🎉

**Comment by MrAE** [comment]
The Travis build will fail for the R package, because it fails the `R CMD check --as-cran`
[00check.log](https://github.com/neurodata/RerF/files/2967111/00check.log)

**Pull request merged**